lua/toggleterm/terminal.lua

```diff
@@ -325,13 +327,31 @@ end
 ---@private
 ---Pass self as first parameter to callback
 function Terminal:__stdout()
-  if self.on_stdout then return function(...) self.on_stdout(self, ...) end end
+  if self.auto_scroll or self.on_stdout then
+    return function(...)
+      if self.auto_scroll then
+        vim.api.nvim_buf_call(self.bufnr, function() vim.cmd('norm G') end)
+      end
+      if self.on_stdout then
+        self.on_stdout(self, ...)
+      end
+    end
+  end
 end
 
 ---@private
 ---Pass self as first parameter to callback
 function Terminal:__stderr()
-  if self.on_stderr then return function(...) self.on_stderr(self, ...) end end
+  if self.auto_scroll or self.on_stderr then
+    return function(...)
+      if self.auto_scroll then
+        vim.api.nvim_buf_call(self.bufnr, function() vim.cmd('norm G') end)
```

Comment:
Same comment as above
```suggestion
         vim.api.nvim_buf_call(self.bufnr, ui.scroll_to_bottom)
```